core/src/main/java/aima/core/environment/map2d/InState.java

```diff
@@ -28,4 +28,16 @@ public int hashCode() {
 	public String toString() {
 		return "In("+getLocation()+")";
 	}
+
+
+	@Override
+	public int compareTo(InState o) {
+		if (getLocation() == null) {
+			return -1;
+		}
+		if (o.getLocation() == null) {
+			return 1;
+		}

```

Comment:
An alternative way of doing this to making InState Comparable would be to create a Comparator like this:

Comparator.<InState>nullsFirst(Comparator.comparing(is -> is.getLocation()));

Which I think makes a little more sense in this case.